### PR TITLE
Add CVZ Fluo dataset

### PR DESCRIPTION
**Pull request by anwai98**
This PR adds a new dataset for cell and nuclei segmentation in fluorescence microscopy images. GTG from my side!

PS: @constantinpape: This dataset is actually very good with very nice annotation tbh. I was too tempted to let this slide. Thanks for reminding. I will work on automating downloading via Synapse later (I found a tutorial on their documentation which seems (and is) super easy actually).

### CODE REVIEW

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py [hunk 1]
```diff
@@ -0,0 +1,167 @@
+"""The CVZ-Fluo dataset contains annotations for cell and nuclei segmentation in
+fluorescence microscopy images.
+
+NOTE: You need to install 'synapseclient`, create an account and authentication token on Synapse,
```
Comment (constantinpape) [inline torch_em/data/datasets/light_microscopy/cvz_fluo.py:4]:
```suggestion
NOTE: You need to install synapseclient`, create an account and authentication token on Synapse,
```

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    """Obtain the CVZ-Fluo dataset.
+
+    Args:
+        path: Filepath to a folder where the downloaded data is saved.
+        download: Whether to download the data if it is not present.
+    """
+    if download:
+        raise NotImplementedError("Automatic download is not implemented for this dataset.")
+
+    data_dir = os.path.join(path, r"Annotation Panel Table.xlsx")
```
Comment (constantinpape) [inline torch_em/data/datasets/light_microscopy/cvz_fluo.py:43]: Why do you need to cast this to a raw string here?

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+    if stain_choice not in ["cell", "dapi"]:
+        raise ValueError(f"'{stain_choice}' is not a valid stain choice.")
+
+    raw_paths = natsorted(
+        glob(os.path.join(path, data_choice, f"*-Crop_{stain_choice.title()}_Png.png"), recursive=True)
+    )
+    label_paths = [p.replace("_Png.png", "_Mask_Png.png") for p in raw_paths]
+
+    neu_label_paths = []
+    for lpath in tqdm(label_paths, desc="Preprocessing labels"):
```
Comment (constantinpape) [inline torch_em/data/datasets/light_microscopy/cvz_fluo.py:85]: I would refactor this in a small helper function, since you are processing the labels.

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        neu_lpath = lpath.replace(".png", ".tif")
+        neu_label_paths.append(neu_lpath)
+        if os.path.exists(neu_lpath):
+            continue
+
+        if not os.path.exists(lpath):  # HACK: some paths have weird spacing nomenclature.
+            lpath = Path(lpath).parent / rf" {os.path.basename(lpath)}"
```
Comment (constantinpape) [inline torch_em/data/datasets/light_microscopy/cvz_fluo.py:92]: Why do we need a raw string here?

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+import imageio.v3 as imageio
+from skimage.measure import label as connected_components
+
+from torch.utils.data import Dataset, DataLoader
+
+import torch_em
+
+from .. import util
+from .neurips_cell_seg import to_rgb
```
Comment (constantinpape) [inline torch_em/data/datasets/light_microscopy/cvz_fluo.py:27]: How many channels does the data have and how exactly do we deal with it?

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py (hunk 2 above)
Comment (anwai98) [reply torch_em/data/datasets/light_microscopy/cvz_fluo.py:43]: With filenames having space between them, they create some issues on Windows. Scripting the paths with spaces as raw string usually avoids this issue!

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py (hunk 3 above)
Comment (anwai98) [reply torch_em/data/datasets/light_microscopy/cvz_fluo.py:85]: Okie

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py (hunk 4 above)
Comment (anwai98) [reply torch_em/data/datasets/light_microscopy/cvz_fluo.py:92]: There are a few labels which are named as `/path/to/ IMAGE.tif`, where there is a weird space infront of the capital "I". This creates issues and the path conventions cannot handle this apparently. So I just handle it by adding a space infront and allowing it as a raw string (due to the same reason as mentioned above)

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py (hunk 5 above)
Comment (anwai98) [reply torch_em/data/datasets/light_microscopy/cvz_fluo.py:27]: The ROIs for cell images have all 3-channels. However for ROIs with DAPI-staining, some images have 1 channel and others 3, which leads to collating issues.

File: torch_em/data/datasets/light_microscopy/cvz_fluo.py (hunk 3 above)
Comment (anwai98) [reply torch_em/data/datasets/light_microscopy/cvz_fluo.py]: This is taken care of now. Thanks!

**Review by constantinpape — APPROVED** [review]
